src/main/java/net/syscon/elite/service/support/LocationProcessor.java

```diff
@@ -125,10 +132,35 @@ private static String newDescription(Location location, boolean preferUserDescri
         }
     }
 
+    /**
+     *
+     * @param locationDescription string to convert
+     * @return new location with correct titlecase
+     *
+     */
     public static String formatLocation(final String locationDescription) {
+        // Handle the possibility of the userDescription being empty
+        if (locationDescription == null) {
+            return null;
+        }
         var description = WordUtils.capitalizeFully(locationDescription);
-        description = RegExUtils.replaceAll(description, "hmp|Hmp", "HMP");
-        description = RegExUtils.replaceAll(description, "yoi|Yoi", "YOI");
-        return description;
+        // Using word boundaries to find the right string ensures we catch the strings
+        // wherever they appear in the description, while also avoiding replacing
+        // the letter sequence should it appear in the middle of a word
+        // e.g. this will not match 'mosaic' even though AIC is one of the abbreviations
+        Pattern pattern = Pattern.compile("\\b(" + String.join("|", ABBREVIATIONS) + ")\\b", Pattern.CASE_INSENSITIVE);
+        Matcher matcher = pattern.matcher(description);
+
+        // There could be more than one abbreviation in a string,
+        // e.g. HMP Moorland VCC Room 1
+        // By using the string buffer and the appendReplacement method
+        // we ensure that all the matching groups are replaced accordingly
+        StringBuffer stringBuffer = new StringBuffer();
```

Comment:
Really minor point but nowadays unless you need synchronisation then favour `StringBuilder` over `StringBuffer`.